services/registry/handlers/maintainers.js

```diff
@@ -16,88 +10,71 @@ module.exports = [
   ),
   fork.post(
     '/v1/packages/package/:namespace([^@]+)@:host/:name/maintainers/:invitee',
-    findInvitee(canWrite(invite))
+    authn.required(invite)
   ),
   fork.del(
     '/v1/packages/package/:namespace([^@]+)@:host/:name/maintainers/:invitee',
-    findInvitee(canWrite(remove))
-  ),
-  fork.post(
-    '/v1/packages/package/:namespace([^@]+)@:host/:name/invitation/:member',
-    packageExists(isNamespaceMember(accept))
-  ),
-  fork.del(
-    '/v1/packages/package/:namespace([^@]+)@:host/:name/invitation/:member',
-    packageExists(isNamespaceMember(decline))
+    authn.required(remove)
   )
 ];
 
 async function maintainers(context, { namespace, host, name }) {
-  const pkg = await Package.objects
-    .get({
-      active: true,
-      name,
-      'namespace.active': true,
-      'namespace.name': namespace
-    })
-    .catch(Package.objects.NotFound, () => null);
+  const [err, maintainers] = await context.storageApi.listPackageMaintainers({
+    namespace,
+    host,
+    name,
+    bearer: context.user ? context.user.name : null,
+    page: Number(context.url.query.page) || 0,
+    status: context.url.query.status
+  }).then(
+    xs => [null, xs],
```

Comment:
I like this. Haskell influence?

Reply:
Haha, thanks! I got it from looking at Golang a while back, tbh.